torch/optim/adam.py

```diff
@@ -84,7 +84,7 @@ def _init_group(
                 # Lazy state initialization
                 if len(state) == 0:
                     state['step'] = (
-                        torch.zeros((1,), dtype=torch.float, device=p.device)
+                        torch.tensor(0.0, dtype=torch.float, device=p.device)
```

Comment:
is this different from zeros(()…)?

Reply:
also, do we still need a conditional here at all if we’re just initializing a scalar tensor on whatever device?

Reply:
ah, nice catch